### PR TITLE
feedback component - remove emotion

### PR DESCRIPTION
**Pull request by katiegoines**
#### Description of changes:
Adding the feedback component to the left menu and removing emotion
Adding 'Edit on GitHub' button to left menu

Staging: https://next-release-feedback-component.d1egzztxsxq9xz.amplifyapp.com/

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Review by timngyn — DISMISSED** [review]

**Comment by timngyn** [comment]
Sorry, didn't mean to approve because I noticed the feedback component doesn't "refresh" when we navigate to other pages

File: src/components/Menu/RepoActions/index.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+function createEditLink(directoryPath) {
+  const safePath = directoryPath
+    .split('/')
+    .map(encodeURIComponent)
+    .join('/');
+  return `https://github.com/aws-amplify/docs/edit/main/src/pages${safePath}/`;
+}
+
+export default function RepoActions({ path }) {
+  console.log(path.path);
```
Comment (hbuchel) [inline src/components/Menu/RepoActions/index.tsx:15]: nit: Just a reminder to remove these console statements before you're ready to merge.

File: yarn.lock
Comment (hbuchel) [inline yarn.lock:1]: The changes to this file probably shouldn't be here?

File: src/components/Icons/IconPencil.tsx
Comment (hbuchel) [inline src/components/Icons/IconPencil.tsx:1]: Really appreciate the icon clean up!

**Review by hbuchel — COMMENTED** [review]
Thanks Katie! Left some feedback/suggestions. Additionally, 
- Can we do anything about the gap between the icon and text so it looks less weird/floaty?
Screenshot 2023-10-31 at 10 40 39 AM

- Maybe we don't animate the "Was this page helpful?" text out of view? I feel like the gap left behind is weird.

File: yarn.lock
Comment (katiegoines) [reply yarn.lock:1]: I refreshed the yarn.lock file but am still seeing a bunch of changes. Any idea why?

File: src/components/Feedback/index.tsx [hunk 2]
```diff
@@ -70,9 +70,9 @@ const Feedback = function Feedback() {
       },
       {
         opacity: 1,
-        transform: 'translate(0)',
-        overflow: 'visible',
-        display: 'block'
+        transform: 'translate(0)'
```
Comment (sowmyadev-kota) [inline src/components/Feedback/index.tsx:73]: Can we please remove the comments if not required.

File: src/components/Menu/RepoActions/index.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+              className="repo-actions-link"
+              aria-label="Edit this page on GitHub"
+            >
+              <IconPencil />
+              Edit on GitHub
+            </Flex>
+          </ExternalLink>
+        </Flex>
+      )}
+      ;
```
Comment (timngyn) [inline src/components/Menu/RepoActions/index.tsx:32]: This semicolon might be a typo? It's being shown in the menu

**Review by jacoblogan — APPROVED** [review]

**Review by hbuchel — COMMENTED** [review]
Looks like the horizonta scroll issue is still there: 
Screenshot 2023-11-02 at 2 12 00 PM

You probably will need to adjust the positioning of the vote response?  Since it's hidden via just opacity it will still take up space. I would probably just nix the animation on this text if it helps 🤷‍♀️

File: src/styles/feedback.scss [hunk 4]
```diff
@@ -13,7 +10,7 @@
   padding-left: 0;
```
Comment (jacoblogan) [inline src/styles/feedback.scss:11]: `text-wrap` is unsupported in FF (and not great support other places) can we use `white-space: nowrap` instead?

**Comment by jacoblogan** [comment]
Invisible github link is clickable before submitting feedback

https://github.com/aws-amplify/docs/assets/7351516/f7759248-886f-4eff-aa3f-2cff32e6eef2

**Review by jacoblogan — DISMISSED** [review]

File: src/styles/feedback.scss [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+.vote-response {
+  flex-direction: column;
+  opacity: 0;
+  overflow: visible;
+  white-space: nowrap;
+  padding-left: 0;
+  font-size: 0.9rem;
+  justify-content: center;
+  gap: 0;
+  pointer-events: none;
```
Comment (hbuchel) [inline src/styles/feedback.scss:15]: Let's find an accessible way to hide the link. `opacity: 0` and `pointer-events: none` mean this is still accessible to screen readers and keyboard navigation. It needs to not be in the DOM, set to display: none, or maybe visibility: hidden

**Comment by reesscot** [comment]
Can we make the padding to the right of the positive feedback button the same as the padding for the negative feedback button?
CleanShot 2023-11-03 at 09 29 19
CleanShot 2023-11-03 at 09 28 53

File: src/styles/feedback.scss (hunk 5 above)
Comment (katiegoines) [reply src/styles/feedback.scss]: Corrected. Thank you

**Review by hbuchel — DISMISSED** [review]
Thanks for the updates, Katie!

**Review by timngyn — APPROVED** [review]

**Review by timngyn — APPROVED** [review]

**Review by hbuchel — APPROVED** [review]

**Review by sowmyadev-kota — APPROVED** [review]

**Pull request merged**